GsaGH/Helpers/GH/Legend/ContourLegendManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Windows.Forms;
+
+using GH_IO.Serialization;
+
+using Grasshopper.Kernel;
+
+using GsaGH.Helpers.GH;
+
+using OasysGH.Components;
+
+namespace GsaGH.Helpers {
+  public class ContourLegendManager {
+    private readonly IContourLegendConfiguration _configuration;
+    private readonly IContourLegend _legend;
+    private readonly LegendMenuBuilder _menuBuilder;
+
+    public ContourLegendManager() {
+      _configuration = new ContourLegendConfiguration();
+      _legend = new ContourLegend(_configuration);
+      _menuBuilder = new LegendMenuBuilder();
+    }
+
+    public ContourLegendManager(
+      IContourLegendConfiguration configuration, IContourLegend legend, LegendMenuBuilder menuBuilder) {
+      _configuration = configuration;
+      _legend = legend;
+      _menuBuilder = menuBuilder;
+    }
+
+    //drawing
+    public void DrawLegend(
+      IGH_PreviewArgs args, string title, string bottomText,
+      List<(int startY, int endY, Color gradientColor)> gradients) {
+      _legend.DrawLegendRectangle(args, title, bottomText, gradients);
+    }
+
+    //menu
+    public ToolStripMenuItem CreateMenu(GH_OasysDropDownComponent component, Action updateUI) {
```

Comment:
they all seem to have the same `Action` ? can't we not set it here without the need of the argument?